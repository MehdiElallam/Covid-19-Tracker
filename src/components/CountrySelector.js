import React, {useEffect, useState} from 'react'
import { useSelector, useDispatch } from 'react-redux'
import Form from 'react-bootstrap/Form';
import { Row, Col } from 'react-bootstrap'
import { fetchWorldCountries } from "../redux/actions/countries"
import { fetchCountryData } from "../redux/actions/countryDetails"
import { fetchCountryHistorical } from "../redux/actions/countryDetails"
import { fetchCountryVaccination } from "../redux/actions/countryVaccination"

import { useParams } from 'react-router-dom'

export default function CountrySelector() {

    const dispatch = useDispatch()
    const worldCountries = useSelector(state => state.worldCountries)
    const app = useSelector(state => state.app)
    const [defaultCountry, setdefaultCountry] = useState('UK')
    const params = useParams();

    useEffect(() => {
        dispatch(fetchWorldCountries());
        dispatch(fetchCountryData('UK'));
        dispatch(fetchCountryVaccination('UK'))

    }, [dispatch])

    const countriesList = worldCountries.loading ? <option>Loading...</option> : worldCountries.countries.map((c,key)=> {
                
        return <option key={key} value={c.country}>{c.country}</option>
    })

    const fetchCountryDetails = (e) => {
        
        const countryCode = e.target.value;
        dispatch(fetchCountryData(countryCode))        
        setdefaultCountry(countryCode)
        fetchCountryHistorical(countryCode)
        dispatch(fetchCountryVaccination(countryCode))

    }

    return (
        <>
            <Row>
                <h2>{app.infos.title}</h2>
            </Row>
            <hr />
            <Row>
                <Col xs={6} md={4}>
                    <Form.Label htmlFor="country"><b>{app.infos.title} in :</b></Form.Label>
                </Col>
                <Col xs={6} md={4}>
                    <Form.Select value={defaultCountry} onChange={fetchCountryDetails} aria-label="Country">
                        {countriesList}
                    </Form.Select>
                </Col>
                
            </Row>
        </>
    )
}
