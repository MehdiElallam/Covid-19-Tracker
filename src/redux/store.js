import {createStore, applyMiddleware, compose} from 'redux'
import thunk from 'redux-thunk'
import rootReducer from './reducers'

const initialState = {}

const middlewares = [thunk]

// const store = createStore(rootReducer, initialState, applyMiddleware(...middlewares),window.devToolsExtension && window.devToolsExtension())

const store = compose(
    applyMiddleware(thunk),
    window.devToolsExtension && window.devToolsExtension(),
)(createStore)(rootReducer, initialState);

export default store;