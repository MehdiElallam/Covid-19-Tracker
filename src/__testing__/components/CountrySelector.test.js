import React from 'react';
import { shallow } from 'enzyme';
import CountrySelector from '../../components/CountrySelector';
import {Provider} from 'react-redux';
import store from "../../redux/store"
import configureStore from 'redux-mock-store';


const mockStore = configureStore([]);

describe('CountrySelector', () => {
      
      let store;
      let component;
      
      beforeEach(() => {
        store = mockStore({
            countries : [],
        });

        component = shallow(
            <Provider store={store}>
                <CountrySelector />
            </Provider>   
        );
    });
  
    // RENDERING TEST :
    it('Should render correctly', () => {
      expect(component).toMatchSnapshot();
    });

});