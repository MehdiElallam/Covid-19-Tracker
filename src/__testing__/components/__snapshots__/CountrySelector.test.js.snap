// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelector Should render correctly 1`] = `ShallowWrapper {}`;

exports[`CountrySelector Should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
